### PR TITLE
Adds UI support for changing the source workflow

### PR DESCRIPTION
**Pull request by saurav-c**
## Describe your changes and why you are making these changes
This PR adds UI support for changing the source workflow.

## Related issue number (if any)
ENG 2181

## Loom demo (if any)
https://www.loom.com/share/a834542eeae7492d81c08636d87b963a

## Checklist before requesting a review
- [ ] I have created a descriptive PR title. The PR title should complete the sentence "This PR...".
- [ ] I have performed a self-review of my code.
- [ ] I have included a small demo of the changes. For the UI, this would be a screenshot or a Loom video.
- [ ] If this is a new feature, I have added unit tests and integration tests.
- [ ] I have run the integration tests locally and they are passing.
- [ ] I have run the linter script locally (See `python3 scripts/run_linters.py -h` for usage).
- [ ] All features on the UI continue to work correctly.
- [ ] Added one of the following CI labels:
    - `run_integration_test`: Runs integration tests
    - `skip_integration_test`: Skips integration tests (Should be used when changes are ONLY documentation/UI)

### CODE REVIEW

File: src/ui/common/src/components/workflows/triggerSourceSelector.tsx [hunk 1]
```diff
@@ -3,34 +3,40 @@ import FormControl from '@mui/material/FormControl';
...
-  // TODO: ENG-2181 Add support for changing source trigger
-  const selected = filteredWorkflows[0].id;
+    setSourceId(selected.id);
+  }, [selected, setSourceId]);
 
   const getMenuItems = () => {
-    return filteredWorkflows.map((workflow) => {
+    return workflows.map((workflow) => {
       return (
+        //@ts-ignore
```
Comment (saurav-c) [inline src/ui/common/src/components/workflows/triggerSourceSelector.tsx:35]: @agiron123 I needed to add this tag here to use `value={workflow}`, but that's now causing a linter error. Unsure what the proper change should be.

**Comment by saurav-c** [comment]
cc @vsreekanti for the UI changes

**Comment by vsreekanti** [comment]
The word `Of` is on a new line for some reason. Let's bring it up to be on the same line as "After Completion". Other than, LGTM!

**Comment by saurav-c** [comment]
@likawind Addressed comments and ran the tests you mentioned

**Review by likawind — APPROVED** [review]
Yay! Let's ship this!

**Pull request merged**